src/transformers/pipelines.py

```diff
@@ -2517,6 +2530,8 @@ def _clean_padding_history(self, generated_tensor) -> List[List[int]]:
             is_previous_pad = False
             for token in sequence:
                 if token == self.tokenizer.pad_token_id:
+                    if self.tokenizer.pad_token_id != self.tokenizer.eos_token_id:
```

Comment:
is this related to this issue: https://github.com/huggingface/transformers/issues/8032?

Reply:
I believe it is not. The previous code from the conversational pipeline was assuming that `eos_token_id` is always equal to `pad_token_id`, and therefore only deleting `eos_token_id` starting from the 2nd consecutive occurence (these are the "padding" eos). Generally, `eos_token_id` is not `pad_token_id` and in that case the padding needs to be removed from the fist `pad_token` on. 

The other issue seems to be related to the generation process itself (creation of incorrect attention masks) - this specific statement is a post-processing step after the generation finishes.